jetty-core/jetty-http2/jetty-http2-server/src/main/java/org/eclipse/jetty/http2/server/internal/HttpStreamOverHTTP2.java

```diff
@@ -615,13 +662,19 @@ public Throwable consumeAvailable()
     @Override
     public void onTimeout(TimeoutException timeout, BiConsumer<Runnable, Boolean> consumer)
     {
+        RecycleState previousState = RecycleState.changeState(_recycle, RecycleState.CANNOT_RECYCLE);
+        if (previousState == RecycleState.RECYCLED)
```

Comment:
```suggestion
        if (previousState.isTerminal())
```

or (I like it less):

```suggestion
        if (previousState != RecycleState.CAN_RECYCLE)
```

Reply:
Done.